src/supermarket/app/controllers/cookbooks_controller.rb

```diff
@@ -308,4 +309,17 @@ def apply_filters
       @cookbooks = @cookbooks.filter_badges(params[:badges])
     end
   end
+
+  def classify_metrics(metric_results)
```

Comment:
Looking at the hash returned by the group_by, I think it's an awful data structure to have travel far from the query. But it's probably still useful if its ugliness is kept within the classify_metrics private method.